### PR TITLE
Remove internal error messages from fault injection HTTP response

### PR DESCRIPTION
**Pull request by tshan2001**
### Summary
Remove internal error messages from fault injection HTTP response

### Implementation details
Instead of returning the error from os/exec execution result, we will use an unified error message `internal error` when we run into issues with running the Linux commands.

After auditing all unit tests results, there's only 1 test case that's impacted by this, that's when we simulate a Linux TC command execution failure and we failed to unmarshal the TC output that was supposed to be a string of a json struct. Instead of returning `failed to unmarshal json`, we will just indicate that we've encountered an internal error, since knowing that we failed to unmarshal the json string won't help the customer and at the same time exposed our underlying implementation.

### Testing
All existing tests are passing

New tests cover the changes: N/A

### Description for the changelog
Remove internal error messages from fault injection HTTP response

### Licensing
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go [hunk 1]
```diff
@@ -42,12 +42,15 @@ import (
...
 const (
+	// Request types
 	startFaultRequestType       = "start %s"
 	stopFaultRequestType        = "stop %s"
 	checkStatusFaultRequestType = "check status %s"
-	invalidNetworkModeError     = "%s mode is not supported. Please use either host or awsvpc mode."
-	faultInjectionEnabledError  = "enableFaultInjection is not enabled for task: %s"
-	requestTimedOutError        = "%s: request timed out"
+	// Error messages
+	internalError              = "internal error"
```
Comment (mye956) [inline ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go:50]: Might be useful to also include the request type in the error message that we respond back to. thoughts? @tshan2001 @xxx0624 @amogh09

Comment (tshan2001) [reply ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go:50]: When the customer curls the endpoint, they would need to specify which fault API they're calling. So they would see something like this: 
```
% curl -X POST ${ECS_AGENT_URI}/fault/v1/network-latency/stop
{"Error":"internal error"}
```

If we add the fault type, it would look like this:
```
% curl -X POST ${ECS_AGENT_URI}/fault/v1/network-latency/stop
{"Error":"network latency: internal error"}
```

Since the type is already specified in the API URL, would it be necessary to include that in the error response?

Comment (mye956) [reply ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go:50]: That's true, I was thinking in the case where there's multiple request coming in and out and how to better track it. I don't too strong of an opinion on this actually

**Review by mye956 — APPROVED** [review]

**Review by harishxr — APPROVED** [review]

File: ecs-agent/tmds/handlers/fault/v1/handlers/handlers_test.go [hunk 2]
```diff
@@ -581,7 +580,7 @@ func generateStartBlackHolePortFaultTestCases() []networkFaultInjectionTestCase
 			name:                 fmt.Sprintf("%s fail duplicate chain", startNetworkBlackHolePortTestPrefix),
 			expectedStatusCode:   500,
```
Comment (xxx0624) [inline ecs-agent/tmds/handlers/fault/v1/handlers/handlers_test.go:581]: Shouldn't this be 400 or 409? @mye956 

In this case, although this means the server can't process the request, this failure is due to customer's configuration that they have injected one same fault.

**Review by xxx0624 — APPROVED** [review]

File: ecs-agent/tmds/handlers/fault/v1/handlers/handlers_test.go (hunk 2 above)
Comment (xxx0624) [reply ecs-agent/tmds/handlers/fault/v1/handlers/handlers_test.go:581]: Checked offline. If second same BHP fault is injected, we will return 200 + running which is expected.

Comment (mye956) [reply ecs-agent/tmds/handlers/fault/v1/handlers/handlers_test.go:581]: Will be addressed in the following PR -> https://github.com/aws/amazon-ecs-agent/pull/4386

TLDR, this edge case should never happen

File: ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go [hunk 3]
```diff
@@ -147,14 +150,14 @@ func (h *FaultHandler) StartNetworkBlackholePort() func(http.ResponseWriter, *ht
 			insertTable = "OUTPUT"
 		}
 
-		cmdOutput, cmdErr := h.startNetworkBlackholePort(ctxWithTimeout, aws.StringValue(request.Protocol), port, chainName,
+		_, cmdErr := h.startNetworkBlackholePort(ctxWithTimeout, aws.StringValue(request.Protocol), port, chainName,
```
Comment (amogh09) [inline ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go:153]: I think we should log the command output. Is it logged elsewhere?
